uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php

```diff
@@ -573,4 +584,53 @@ public static function getRemainder($params){
     return $results;
   }
 
+  /**
+   * Returns formatted results for getting the breakdown for a period entitlement
+   * i.e all of the leave balance changes given a leavePeriodEntitlement ID or (ContactID + periodId)
+   *
+   * @param array $params
+   *
+   * @return array
+   */
+  public function getBreakdown($params) {
+
+    $leavePeriodEntitlements = [];
+
+    if(!empty($params['entitlement_id'])) {
+      $leavePeriodEntitlements[] = self::findById($params['entitlement_id']);
+
+    }
+
+    if(!empty($params['contact_id']) && !empty($params['period_id'])){
+      $leavePeriodEntitlements = self::getPeriodEntitlementsForContact($params['contact_id'], $params['period_id']);
+    }
+
+    $leaveBalanceTypeIdOptionsGroup = [];
+    $leaveBalanceChangeTypeIdOptions = LeaveBalanceChange::buildOptions('type_id');
+    foreach($leaveBalanceChangeTypeIdOptions as $key => $label){
+      $leaveBalanceTypeIdOptionsGroup[$key] = [
+        'id' => $key,
+        'label' => $label,
+        'value' => CRM_Core_Pseudoconstant::getName('LeaveBalanceChange', 'type_id', $key)
+      ];
+    }
+
+    $results = [];
+    $returnExpired = !empty($params['expired']) ? true : false;
+    $i = 0;
+    foreach($leavePeriodEntitlements as $leavePeriodEntitlement){
```

Comment:
For PhpStorm, check this help page: https://www.jetbrains.com/help/phpstorm/2016.2/code-style-php.html

Take a look at the "Set from" part. You'll need to set from Drupal.